framework/src/utils/MooseObjectParameterName.C

```diff
@@ -0,0 +1,126 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+// MOOSE includes
+#include "MooseObjectParameterName.h"
+
+// STL includes
+#include <iostream>
+
+MooseObjectParameterName::MooseObjectParameterName(const MooseObjectName & obj_name, std::string param) :
+    MooseObjectName(obj_name),
```

Comment:
MooseObjectParameter name does inherit from MooseObjectName, unfortunately the logic for parsing the string containing only the object name is different that object and parameter name, unless you can come up with something better. I am not as experience with parsing as you.